src/client.js

```diff
@@ -4644,8 +4644,11 @@ function setupCallEventHandler(client) {
     // This happens quite often, eg. replaying sync from storage, catchup sync
     // after loading and after we've been offline for a bit.
     let callEventBuffer = [];
-    function onSync(state) {
-        if (state === "SYNCING") {
+    function evaluateEventBuffer() {
+        if (client.getSyncState() === "SYNCING") {
+            // don't process any events until they are all decrypted
+            if (callEventBuffer.some((e) => e.isBeingDecrypted())) return;
```

Comment:
if for some events we never get the keys, won't calls completely stop working?

Reply:
`decrypted` is a bit misleading and actually means, "finished trying to decrypt" so will fire when we've either decrypted it or given up trying.

Reply:
right, but some events in the buffer might never get decrypted, even though all the `m.call` events in the buffer are already decrypted. As the buffer is never emptied in this case, I'm wondering if this would ever resolve itself. Maybe I'm missing something.

Reply:
> `decrypted` is a bit misleading and actually means, "finished trying to decrypt" so will fire when we've either decrypted it or given up trying.

Oh, it also fires when giving up, ok :+1: